src/rust/engine/process_execution/src/nailgun/nailgun_pool.rs

```diff
@@ -331,16 +417,17 @@ impl NailgunProcess {
           port: port,
           fingerprint: nailgun_server_fingerprint,
           name: name,
-          handle: Arc::new(Mutex::new(child)),
+          handle: Arc::new(parking_lot::Mutex::new(child)),
           build_id: build_id,
         })
       })
+      .to_boxed()
   }
 }
 
 impl Drop for NailgunProcess {
   fn drop(&mut self) {
-    debug!("Exiting nailgun server process {:?}", self);
+    // TODO we should probaly use the nails client to call the ng-kill nail?
```

Comment:
`probably`

But: I don't think that that is necessary... we know that we won't have concurrency, so we should not have any need for "clean shutdown".

Reply:
Makes sense to me! I have seen orphaned nailgun processes during my debugging, but those might result from sending SIGINT to pants while running or something like that.